.github/workflows/compatability-tests.yaml

```
@@ -0,0 +1,288 @@
+        run: sbt sonnets-311/package
+
+      - name: Generate uniquifying value
+        id: unique
+        run: echo "value=$RANDOM" >> $GITHUB_OUTPUT
+
+      - name: Start lakeFS for Spark tests
+        uses: ./.github/actions/bootstrap-test-lakefs
+        with:
+          compose-directory: test/spark
+        env:
+          REPO: treeverse
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          LAKEFS_DATABASE_TYPE: postgres
+          LAKEFS_BLOCKSTORE_TYPE: s3
+          LAKEFS_BLOCKSTORE_S3_CREDENTIALS_ACCESS_KEY_ID: ${{ secrets.ESTI_AWS_ACCESS_KEY_ID }}
+          LAKEFS_BLOCKSTORE_S3_CREDENTIALS_SECRET_ACCESS_KEY: ${{ secrets.ESTI_AWS_SECRET_ACCESS_KEY }}
+
+      - name: Setup lakeFS for tests
+        working-directory: test/spark
+        run: ./setup-test.sh
+
+      - name: Retrieve client
+        uses: actions/download-artifact@v3
+        with:
+          name: client-assembly
+          path: clients/hadoopfs/target
+
+      - name: Test lakeFS S3 with Spark 3.x thick client
+        timeout-minutes: 8
+        env:
+          JARS: clients/hadoopfs/
+          STORAGE_NAMESPACE: s3://esti-system-testing/compatibility/${{ github.run_number }}-spark3-client/${{ steps.unique.outputs.value }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.ESTI_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ESTI_AWS_SECRET_ACCESS_KEY }}
+          USE_DIRECT_ACCESS: "true"
+          REPOSITORY: thick-client-test
+          SONNET_JAR: sonnets-311/target/sonnets-311/scala-2.12/sonnets-311_2.12-0.1.0.jar
+        working-directory: test/spark
+        run: ./run-test.sh
+
+      - name: lakeFS Logs on Spark with client failure
+        if: ${{ failure() }}
+        continue-on-error: true
+        working-directory: test/spark
+        run: docker-compose logs --tail=15000 lakefs
+
+  compatibility-checks-server:
+    name: Test lakeFS server FileSystem compatibility
+    needs: [gen-code, deploy-image]
+    strategy:
+      fail-fast: false
+      matrix:
+        # Removing a version from this list means the current lakeFS is no longer compatible with
+        # that Hadoop lakeFS client version.
+        client_version: [ 0.1.10, 0.1.11, 0.1.12 ]
+    runs-on: ubuntu-20.04
+    env:
+      CLIENT_VERSION: ${{ matrix.client_version }}
+      TAG: ${{ needs.deploy-image.outputs.tag }}
+      IMAGE_ID: ${{ needs.deploy-image.outputs.image_id }}
+      REPO: treeverse
+      SPARK_TAG: 3
+    steps:
+      - name: Check-out code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: lakefs-image
+          path: /tmp
+
+      - name: Load Docker image
+        run: |
+          docker load --input /tmp/lakefs.tar
+          docker image ls -a
```

no need to list images
```suggestion
        run: docker load --input /tmp/lakefs.tar
```

---

Done.